library/ceph_key.py

```diff
@@ -0,0 +1,422 @@
+#!/usr/bin/python
+# Copyright 2018, Red Hat, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.1',
+    'status': ['preview'],
+    'supported_by': 'community'
+}
+
+DOCUMENTATION = '''
+---
+module: ceph_key
+
+author: Sebastien Han <seb@redhat.com>
+
+short_description: Manage Cephx key(s)
+
+version_added: "2.6"
+
+description:
+    - Manage CephX creation, deletion and updates.
+    It can also list and get information about key(s)
+options:
+    cluster:
+        description:
+            - The ceph cluster name.
+        required: false
+        default: ceph
+    name:
+        description:
+            - name of the CephX key
+        required: true
+    state:
+        description:
+            - present|absent|list|update|info
+        required: true
+        choices: ['present', 'absent', 'list', 'update', 'info']
+        default: list
+    caps:
+        description:
+            - CephX key capabilities
+        required: false
+    secret:
+        description:
+            - key's secret value
+        required: false
+        default: None
+    containerized:
+        description:
+            - Weither or not this is a containerized cluster. The value is
+            assigned or not depending on how the playbook runs.
+        required: false
+        default: None
+'''
+
+EXAMPLES = '''
+- name: create cephx key
+  ceph_key:
+    name: "my_key"
+    state: present
+    caps: "{{ caps }}"
+
+- name: update cephx key
+  ceph_key:
+    name: "my_key"
+    state: update
+    caps: "{{ caps }}"
+
+- name: delete cephx key
+  ceph_key:
+    name: "my_key"
+    state: absent
+
+- name: info cephx key
+  ceph_key:
+    name: "my_key""
+    state: info
+
+- name: list cephx keys
+  ceph_key:
+    state: list
+'''
+
+RETURN = '''#  '''
+
+from ansible.module_utils.basic import AnsibleModule
+import datetime
+import os
+import struct
+import time
+import base64
+
+
+def fatal(message, module):
+    '''
+    Report a fatal error and exit
+    '''
+
+    if module:
+        module.fail_json(msg=message, rc=1)
+    else:
+        raise(Exception(message))
+
+
+def key_mode(file_path):
+    '''
+    Change mode file for a CephX key
+    Problem, how do to do on containerized deployment?
+    '''
+    os.chmod(file_path)
```

Comment:
`chmod` takes two arguments, here it is using just one

Reply:
What calls `key_mode`? In the past with ceph-deploy we've run into trouble with running chmod on keys after they are already written to disk with looser perms, see CVE-2015-3010 and CVE-2015-4053

Reply:
This is a placeholder for now, I'm hoping to get this merged at some point: https://github.com/ceph/ceph/pull/20983.

So I started this but this won't quite work on containers.